### PR TITLE
[CELEBORN-511][IMPROVE] Move onTrim tag to StorageManager to avoid frequent trim action

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Currently, if we call trim. in StorageManager, it still just add it to actionService, then return, the flag will directly set to false, then add a new trim action, here we should directly execute the onTrim in StorageManager.

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@FMX @waitinfuture

**Review by FMX — CHANGES_REQUESTED** [review]
I think it's not a good idea because the onTrim action should run as fast as possible to ensure that the memory manager is working as designed.
I think adding a variable to indicate that Storage Manager is executing trim actions would be better.

**Comment by AngersZhuuuu** [comment]
@FMX Update, how about current?

**Review by FMX — APPROVED** [review]
LGTM with minor comment.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala [hunk 1]
```diff
@@ -677,11 +678,19 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
   override def onResume(moduleName: String): Unit = {}
 
   override def onTrim(): Unit = {
-    actionService.submit(new Runnable {
-      override def run(): Unit = {
-        flushFileWriters()
-      }
-    })
+    if (!trimInProcess.get()) {
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala]: Done

**Comment by AngersZhuuuu** [comment]
ping @FMX

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala [hunk 2]
```diff
@@ -677,11 +678,18 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
   override def onResume(moduleName: String): Unit = {}
 
   override def onTrim(): Unit = {
-    actionService.submit(new Runnable {
-      override def run(): Unit = {
-        flushFileWriters()
-      }
-    })
+    if (trimInProcess.compareAndSet(false, true)) {
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:681]: Shall we also add the similar logic in ChannelsLimiter? Maybe we should put trimInProcess in MemoryPressureListener then.

Comment (FMX) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:681]: ChannelLimiter's onTrim isn't blocking operation.

**Comment by AngersZhuuuu** [comment]
Any more suggestion?

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala (hunk 2 above)
Comment (waitinfuture) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:681]: > ChannelLimiter's onTrim isn't blocking operation.

I'm not sure about the cost of frequently calling  ChannelLimiter.onTrim

Comment (AngersZhuuuu) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:681]: > As far as I know that this PR didn't change channel limiters' frequency.

I think the last pr changed this, we need a new pr to discuss and change this.

Comment (AngersZhuuuu) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:681]: I will merge this first, then I will find some unhealthy workers and then provide a fire flame about the cost during trim action. WDYT?

Comment (waitinfuture) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:681]: > I will merge this first, then I will find some unhealthy workers and then provide a fire flame about the cost during trim action. WDYT?

SGTM, thanks!

**Comment by AngersZhuuuu** [comment]
截屏2023-04-14 下午12 03 17

**Pull request merged**